src/color_wheel.cpp

```diff
@@ -571,4 +667,33 @@ void ColorWheel::dropEvent(QDropEvent* event)
     }
 }
 
+void ColorWheel::clearHarmonies()
+{
+    p->ring_editors.clear();
+    p->current_ring_editor = -1;
+    Q_EMIT harmonyChanged();
+    update();
+}
+
+void ColorWheel::addHarmony(double hue_diff, bool editable, int symmetric_to, int opposite_to)
+{
+    hue_diff = normalize(hue_diff);
+    if (symmetric_to >= 0 && opposite_to >= 0)
+        throw std::runtime_error("incorrect call to addHarmony: harmony cannot be both symmetric and opposite");
+    int harmony_count = p->ring_editors.size();
+    if (symmetric_to >= harmony_count || opposite_to >= harmony_count)
+        throw std::runtime_error("incorrect call to addHarmony: harmony number out of range");
+    p->ring_editors.emplace_back(hue_diff, editable, symmetric_to, opposite_to);
```

Comment:
I think hue_diff should be adjusted for simmetric/opposite ones (so it starts with the right value)

Perhaps it's better to have different `addHarmony` functions to make it clearer how it works (for example):
* `addHarmony(hue_diff, editable)`
* `addSimmetricHarmony(relative_to, editable)`
* `addOppositeHarmony(relative_to, editable)`